### ISSUE TITLE
gitin_0.1.0_linux_amd64.tar.gz release isn't a self-contained static binary, has runtime dependency on libssl.so.1.1

### ISSUE THREAD

**Issue by fcostin** [isacikgoz/gitin#12]
the linux release `gitin_0.1.0_linux_amd64.tar.gz`  is not self-contained, has runtime dependencies on dynamic libraries (e.g. libssl.so.1.1 )

### steps to reproduce

0. obtain a standard test environment without dev dependencies, e.g. a debian jessie slim container

1. download binary release https://github.com/isacikgoz/gitin/releases/download/v0.1.0/gitin_0.1.0_linux_amd64.tar.gz

2. untar the `gitin`

3. add the `gitin` to the PATH

4. run `gitin --version`

### expected behaviour

gitin exits successfully after displaying version number

### observed behaviour

```
gitin: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory
```

### dockerfile to reproduce

`docker build -t gitin_static:dev -f test.Dockerfile .`

where `test.Dockerfile` contains

```
FROM debian:jessie-20190122-slim

RUN apt update
RUN apt install -y wget

WORKDIR /tmp
RUN wget --quiet https://github.com/isacikgoz/gitin/releases/download/v0.1.0/gitin_0.1.0_linux_amd64.tar.gz
RUN tar -xvzf gitin_0.1.0_linux_amd64.tar.gz
RUN mkdir -p /tmp/bin
RUN mv gitin /tmp/bin/gitin
ENV PATH="$PATH":/tmp/bin/

RUN gitin --version
```

### proposed cause

git2go links to libgit2, which in turn links to various dependencies for ssh support, http support, etc. it seems like these latter dependencies tend to be linked dynamically, even if you use the scripts and documentation provided by git2go to do a "static" build of git2go.

### suggested fix

if the gitin application does not actually require ssh or http support, you can customise build flags of libgit2 when compiling git2go to disable linking to these unnecessary dependencies.

libgit2 build flags that can be customised are here: https://github.com/libgit2/libgit2/blob/master/CMakeLists.txt#L58

i suggest disabling USE_SSH, USE_HTTPS, and USE_EXT_HTTP_PARSER (these are all educated guesses based on the option name)

one way to do this is by modifying: https://github.com/libgit2/git2go/blob/master/script/build-libgit2-static.sh

pass these additional flags to `cmake`: -DUSE_SSH=OFF -DUSE_HTTPS=OFF -DUSE_EXT_HTTP_PARSER=OFF after the `-DBUILD_SHARED_LIBS=OFF` flag.

then, to build `git2go` run this modified `script/build-libgit2-static.sh` .

the `gitin` application should still link against this new static git2go library if built using `--tags static` (as it looks like it is currently built)

(this fix may not be 100% accurate, it's from memory -- i spent a few hours earlier today trying to figure how to do self-contained static builds for my own application that depends on git2go)

**Comment by isacikgoz** [comment]
Thanks for the comprehensive feedback @fcostin I really appreciate it. I will try to handle this issue with your approach.

**Comment by isacikgoz** [comment]
I added these options so that it won't require such external libs in the runtime. I will build with this configuration for the next release.

```patch
...
+      -DUSE_EXT_HTTP_PARSER=OFF \
+      -DUSE_HTTPS=OFF \
+      -DUSE_NSEC=OFF \
+      -DUSE_SSH=OFF \
...
```

**Comment by fcostin** [comment]
Thank you! Hope that the adjusted build still works. I'll be away from my
computer for a few days but I'm happy to test a new build next week

**Comment by isacikgoz** [comment]
I've just fixed it and it's merged to master. Please let me know if you still the issue. Also, feel free to open a new issue.

**Comment by fcostin** [comment]
confirm this is fixed in the latest release ( https://github.com/isacikgoz/gitin/releases/download/v0.1.0/gitin_0.1.0_linux_amd64.tar.gz ) , no action required. :+1:
